CarpetX/src/io_openpmd.cxx

```diff
@@ -1155,38 +1154,50 @@ void carpetx_openpmd_t::InputOpenPMD(const cGH *const cctkGH,
         assert(cactus_dk > 0);
         assert(cactus_np > 0);
         for (int vi = 0; vi < numvars; ++vi) {
-          CCTK_REAL *const cactus_var_ptr =
-              groupdata.data.at(tl).data() + vi * cactus_np;
+          void *const cactus_var_ptr =
+              groupdata.data.at(tl).data_at(vi * cactus_np);
           if (input_ghosts || intbox == extbox) {
-#if OPENPMDAPI_VERSION_GE(0, 15, 0)
-            record_components.at(vi).loadChunkRaw(cactus_var_ptr, start, count);
-#else
-            record_components.at(vi).loadChunk(
-                openPMD::shareRaw(cactus_var_ptr), start, count);
+#if !OPENPMDAPI_VERSION_GE(0, 15, 0)
+#define loadChunkRaw(ptr, start, count)                                        \
+  loadChunk(openPMD::shareRaw(ptr), start, count)
 #endif
-
+            switch (cgroup.vartype) {
+            case CCTK_VARIABLE_REAL:
+              record_components.at(vi).loadChunkRaw(
+                  static_cast<CCTK_REAL *>(cactus_var_ptr), start, count);
+              break;
+            case CCTK_VARIABLE_INT:
+              record_components.at(vi).loadChunkRaw(
+                  static_cast<CCTK_INT *>(cactus_var_ptr), start, count);
+              break;
+            case CCTK_VARIABLE_COMPLEX:
+              record_components.at(vi).loadChunkRaw(
+                  static_cast<CCTK_COMPLEX *>(cactus_var_ptr), start, count);
+              break;
+            default:
+              assert(0 && "Unexpected variable type");
```

Comment:
I'd call `CCTK_ERROR` here because it can't be disabled by compiler options.

Reply:
this is not user triggerable (via parfile input) and the list is fixed by Cactus (not user or CarpetX code), so I'd rather keep the assert, if possible.